src/Provider/PointCloudProvider.js

```diff
@@ -174,6 +175,23 @@ export default {
         layer.update = PointCloudProcessing.update;
         layer.postUpdate = PointCloudProcessing.postUpdate;
 
+        // override the default method, since updated objects are metadata in this case
+        layer.getObjectToUpdateForAttachedLayers = (meta) => {
```

Comment:
It should move this function in the processing, and assign it as above (same think  for 3dtiles)
```js
layer.getObjectToUpdateForAttachedLayers = PointCloudProcessing.getObjectToUpdateForAttachedLayers;
```

the `provider.preprocess` it looks like a layer's constructor. `3dTiles` and `pointCloud` should inherit `geometryLayer` and overload its processing functions (if that is necessary)

Reply:
> It should move this function in the processing, and assign it as above

Where? Processing have no code that run once for a layer, while Provider have a preprocess step.
And processing code should try to be as independant as possible from providers so moving these functions that are highly dependent on the protocol/provider makes very little sense.

One of my mid-term goal is to have a single processing module for pointclouds, that would work with all kind of pointcloud sources.
To be able to do this, the provider of the said sources has to format/prepare the layer to meet the expectations of this processing module.

Basically: the processing code defines a contract, and will work with every layer that implements it.

> the provider.preprocess it looks like a layer's constructor 

Can you explain what you mean?

> 3dTiles and pointCloud should inherit geometryLayer and overload its processing functions (if that is necessary)

I assume that you meant 3dtiles *layer* and pointcloud *layer*?
What would be the benefits of inheriting + overriding `GeometryLayer` over the current situation?

(as there are no processing functions in GeometryLayer so I don't understand this either)

Reply:
> as there are no processing functions in GeometryLayer so I don't understand this either

`geometryLayer.getObjectToUpdateForAttachedLayers` is processing function

Reply:
Ok I don't want to change this because I simply don't agree.

The problem currently is not this code, but the fact that Provider and Processing have strong coupling.

I and others are working on improving this and I would appreciate that you don't undermine this effort by bringing this topic on each review.